runway/module/__init__.py

```diff
@@ -60,67 +24,169 @@ def run_module_command(cmd_list, env_vars):
-    # https://docs.npmjs.com/cli/ci#description
-    with open(os.devnull, 'w') as fnull:
-        if ((os.path.isfile(os.path.join(path,
-                                         'package-lock.json')) or
-             os.path.isfile(os.path.join(path,
-                                         'npm-shrinkwrap.json'))) and
-                subprocess.call(
-                    [NPM_BIN, 'ci', '-h'],
-                    stdout=fnull,
-                    stderr=subprocess.STDOUT
-                ) == 0):
-            return True
-    return False
-
-
-def run_npm_install(path, options, context):
-    """Run npm install/ci."""
-    # Use npm ci if available (npm v5.7+)
-    if options.get('skip_npm_ci'):
-        LOGGER.info("Skipping npm ci or npm install on %s...",
-                    os.path.basename(path))
-    elif context.env_vars.get('CI') and use_npm_ci(path):  # noqa
-        LOGGER.info("Running npm ci on %s...",
-                    os.path.basename(path))
-        subprocess.check_call([NPM_BIN, 'ci'])
-    else:
-        LOGGER.info("Running npm install on %s...",
-                    os.path.basename(path))
-        subprocess.check_call([NPM_BIN, 'install'])
-
-
-class RunwayModule(object):
+class RunwayModule(object):  # noqa pylint: disable=too-many-instance-attributes
     """Base class for Runway modules."""
 
-    def __init__(self, context, path, options=None):
+    def __init__(self, context, path, runway_file_options=None):
         """Initialize base class."""
+        self.name = os.path.basename(path)
+
         self.context = context
 
+        self.folder = RunwayModuleFolder(path)
+
+        # it would be good to remove the need for sub-classes to refer
+        #  to this directly, and have them rely on 'folder' instead
         self.path = path
 
-        if options is None:
-            self.options = {}
-        else:
-            self.options = options
+        environments_options = better_dict_get(runway_file_options, 'environments', {})
+        self.environment_options = environments_options.get(context.env_name)
+        if self.environment_options is not None:
+            # it will almost always be a 'dict', but some modules (like serverless) support
+            #  a boolean, which indicates we want the module, and we have no values to set
+            if isinstance(self.environment_options, bool):
+                if self.environment_options:
+                    self.environment_options = {}
+                else:
+                    self.environment_options = None
+
+        self.module_options = better_dict_get(runway_file_options, 'options', {})
 
-    # the rest of these 'abstract' methods must have names which match
-    #  the commands defined in `cli.py`
+        # should global options be in the context instead?
+        self.project_options = {}
+        self.project_options['skip_npm_ci'] = runway_file_options.get('skip_npm_ci')
```

Comment:
There were a couple of mistakes in retrospect here: having skip-npm-ci be a deployment option and not placing `skip_npm_ci` in the nested 'options' in the module's options (what I above call the `module_custom_options`).

Let's start deprecating them, by:

1. Updating the docs
2. Detecting their use, warning that they've been deprecated and will be removed in 1.0, and set the custom module options on behalf of the user for now